sdk/core/azure-core-amqp/src/amqp/private/message_sender_impl.hpp

```diff
@@ -5,34 +5,41 @@
 
 #include "azure/core/amqp/message_sender.hpp"
 #include "claims_based_security_impl.hpp"
+#include "link_impl.hpp"
 #include <azure_uamqp_c/message_sender.h>
 #include <tuple>
 
+template <> struct Azure::Core::_internal::UniqueHandleHelper<MESSAGE_SENDER_INSTANCE_TAG>
+{
+  static void FreeMessageSender(MESSAGE_SENDER_HANDLE obj);
+
+  using type
+      = Azure::Core::_internal::BasicUniqueHandle<MESSAGE_SENDER_INSTANCE_TAG, FreeMessageSender>;
+};
+
 namespace Azure { namespace Core { namespace Amqp { namespace _detail {
```

Comment:
This one I'd also like to see if it is possible to have it in unnamed namespace (i.e. without linker symbols exported), but I am not going to look for this PR. I think it should be possible to move it to unnamed namespace, but I don't experimented/looked at the code enough to say so definitively.